daemons/pacemakerd/pacemakerd.c

```diff
@@ -185,27 +185,22 @@ static void
 remove_core_file_limit(void)
 {
     struct rlimit cores;
-    int rc = getrlimit(RLIMIT_CORE, &cores);
 
-    if (rc < 0) {
-        crm_warn("Cannot determine current maximum core file size: %s",
-                 strerror(errno));
-        return;
-    }
-
-    if ((cores.rlim_max == 0) && (geteuid() == 0)) {
-        cores.rlim_max = RLIM_INFINITY;
-    } else {
-        crm_info("Maximum core file size is %llu bytes",
-                 (unsigned long long) cores.rlim_max);
-    }
-    cores.rlim_cur = cores.rlim_max;
-
-    rc = setrlimit(RLIMIT_CORE, &cores);
-    if (rc < 0) {
-        crm_warn("Cannot raise system limit on core file size "
-                 "(consider doing so manually)");
+    if (getrlimit(RLIMIT_CORE, &cores) == 0) {
+        if ((cores.rlim_max == 0) && (geteuid() == 0)) {
+            cores.rlim_max = RLIM_INFINITY;
+        } else {
+            crm_trace("Maximum core file size is %llu bytes",
+                      (unsigned long long) cores.rlim_max);
+        }
+        cores.rlim_cur = cores.rlim_max;
+        if (setrlimit(RLIMIT_CORE, &cores) == 0) {
+            return;
+        }
     }
+    crm_notice("Cannot raise system limit on core file size "
+               "(consider doing so manually): %s",
+               strerror(errno));
 }
 
```

Comment:
RLIM_INFINITY is 0 ... still a misleading message but at trace level I'm not too worried about it

agreed we should bail if cur is already max, will do in followup

Reply:
Whoops, I misread, RLIM_INFINITY is ~0, so your comment stands

Reply:
So you're suggesting we raise the hard limit to RLIM_INFINITY even if it's currently nonzero?

I suppose the question is what the justification for overriding the setting is. It was added in 56c37f3a0 without much explanation. I guess the thinking was that if the admin explicitly set a particular (finite) limit, we should respect that, but if it's disabled entirely, that's more likely a system or user default that shouldn't necessarily apply to any specific daemon.

I don't have a strong opinion on it one way or the other, but I lean to interfering with admin settings as little as possible.

Reply:
> So you're suggesting we raise the hard limit to RLIM_INFINITY even if it's currently nonzero?

No. We just might still print out what we are setting it to instead of being silent about it in the
case it is disabled.
Of course in the root case going for unlimited in any case could be considered as well.
But as you write below - if something finite is set it is probably on purpose.

> 
> I suppose the question is what the justification for overriding the setting is. It was added in [56c37f3](https://github.com/ClusterLabs/pacemaker/commit/56c37f3a09a5d91bb766ef10f7275e95c2491fb9) without much explanation. I guess the thinking was that if the admin explicitly set a particular (finite) limit, we should respect that, but if it's disabled entirely, that's more likely a system or user default that shouldn't necessarily apply to any specific daemon.
> 
> I don't have a strong opinion on it one way or the other, but I lean to interfering with admin settings as little as possible.